pkg/tests/ossm/bug_istiopods_test.go

```diff
@@ -15,79 +15,66 @@
-	bookinfo.Install(false)
+		t.Cleanup(func() {
+			DeleteNamespaces(t, data)
+			oc.RecreateNamespace(t, meshNamespace)
+			oc.ApplyString(t, meshNamespace, smmr)
+		})
 
-	t.Run("smcp_test_istio_pod_probes_failure", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-		log.Log.Info("Testing: Istio Pod get stuck with probes failure after restart")
-		log.Log.Infof("Create %d new namespaces", numberOfNamespaces)
-		util.Shell(fmt.Sprintf(env.GetRootDir()+"/scripts/smmr/create_members.sh %d", numberOfNamespaces))
-		log.Log.Info("Namespaces created...")
-		rand.Seed(time.Now().UnixNano())
-		// Random number of deletes for the pod between 4 and 10
-		randId := rand.Intn(10-4) + 4
-		log.Log.Info("Random number of deletes: ", randId)
-		count := 0
-		for count < randId {
-			log.Log.Info("*** Get the istiod pod name")
-			istiod, _ := util.GetPodName(`istio-system`, `app=istiod`)
-			log.Log.Info("*** Delete Istio pod: ", istiod)
-			util.Shell(`oc delete pod %s -n istio-system`, istiod)
-			deleted, _ := util.CheckPodDeletion(`istio-system`, `app=istiod`, istiod, 60)
-			if deleted {
-				log.Log.Info("*** Istiod pod deleted: ", istiod)
-				running := util.CheckPodRunning(`istio-system`, `app=istiod`)
-				if running == nil {
-					log.Log.Info("*** New Istiod pod is running")
-					ready, _ := util.CheckPodReady(`istio-system`, `app=istiod`, 10)
-					if ready {
-						log.Log.Info("*** New Istiod pod is ready")
-					} else {
-						istiod, _ = util.GetPodName(`istio-system`, `app=istiod`)
-						// Get events that are not type = Normal from the pod
-						event, _ := util.Shell(`kubectl get events  -n istio-system --field-selector type!=Normal,involvedObject.name=%s |tail -1`, istiod)
-						if strings.Contains(event, "Readiness probe failed") {
-							t.Fatalf("****** Istio pod is not running and fail because of readiness probe failure: %s", event)
-						} else if event == "No resources found in istio-system namespace." {
-							log.Log.Info("*** No eventss for pod: ", istiod)
-						} else {
-							t.Errorf("*** Istio pod is not running but is not failing because of readiness probe failure: %s", event)
-						}
-					}
-					count++
-				}
-			} else {
-				t.Errorf("*** Istiod pod is not deleted: %s", istiod)
-			}
+		t.LogStep("Create Namespaces and SMMR")
+		CreateNamespaces(t, data)
+		UpdateSMMRMultipleNamespaces(t, meshNamespace, smmr_multiple_members, data)
+
+		t.LogStep("Delete Istio pod 10 times and check that it is running and ready after the deletions")
+		for i := 0; i < 10; i++ {
+			oc.DeletePod(t, pod.MatchingSelector("app=istiod", meshNamespace))
+			oc.WaitPodRunning(t, pod.MatchingSelector("app=istiod", meshNamespace))
+			oc.WaitPodReady(t, pod.MatchingSelector("app=istiod", meshNamespace))
 		}
-		log.Log.Info("*** Test finished without errors after deletion of pods")
 	})
 }
+
+func UpdateSMMRMultipleNamespaces(t test.TestHelper, ns string, yaml string, data interface{}) {
```

Comment:
If we only need to call these three functions in the ossm package, local declaration of functions should start with lower case.
"UpdateSMMRMultipleNamespaces", "CreateNamespaces", "DeleteNamespaces"

Reply:
Done, but I have to change the names to createMultipleNamespaces, deleteMultipleNamespaces. Because createNamespaces is declared in setup.go